addons/account/tests/test_tax.py

```diff
@@ -92,29 +162,46 @@ def compare_values(results, expected_values, rounding):
-        compute_kwargs = compute_kwargs or {}
-        quantity = evaluation_context_kwargs.pop('quantity', 1)
-        evaluation_context = taxes._eval_taxes_computation_prepare_context(price_unit, quantity, **evaluation_context_kwargs)
-        taxes_computation = taxes._prepare_taxes_computation(taxes._convert_to_dict_for_taxes_computation(), **compute_kwargs)
-        results = taxes._eval_taxes_computation(taxes_computation, evaluation_context)
-        is_round_globally = evaluation_context_kwargs.get('rounding_method') == 'round_globally'
-        rounding = 0.000001 if is_round_globally else 0.01
-        compare_values(results, expected_values, rounding)
-
-        # Check the reverse in case of round_globally.
-        if is_round_globally:
-            evaluation_context = taxes._eval_taxes_computation_prepare_context(results['total_excluded'], quantity, **{
-                **evaluation_context_kwargs,
-                'reverse': True,
-            })
-            results = taxes._eval_taxes_computation(taxes_computation, evaluation_context)
-            compare_values(results, expected_values, rounding)
-            delta = sum(x['tax_amount_factorized'] for x in results['tax_values_list'] if x['price_include'])
-            self.assertEqual(
-                float_round(results['total_excluded'] + delta, precision_rounding=rounding),
-                float_round(price_unit, precision_rounding=rounding),
-            )
+        params = test['params']
+        if params['test'] == 'taxes_computation':
+            is_round_globally = params['is_round_globally']
+            rounding = 0.000001 if is_round_globally else 0.01
+
+            for results_key in ('py_results', 'js_results'):
+                with self.subTest(results_key=results_key):
+                    if results_key not in test:
+                        continue
+
+                    results = test[results_key]
+                    compare_taxes_computation_values(results['results'], test['expected_values'], rounding)
+
+                    # Check the reverse in case of round_globally.
+                    if is_round_globally:
+                        compare_taxes_computation_values(results['reverse_results'], test['expected_values'], rounding)
+                        delta = sum(x['tax_amount_factorized'] for x in results['reverse_results']['tax_values_list'] if x['price_include'])
+
+                        self.assertEqual(
+                            float_round(results['reverse_results']['total_excluded'] + delta, precision_rounding=rounding),
+                            float_round(params['price_unit'], precision_rounding=rounding),
+                        )
+        elif params['test'] == 'adapt_price_unit_to_another_taxes':
+            for results_key in ('py_results', 'js_results'):
+                with self.subTest(results_key=results_key):
+                    if results_key not in test:
+                        continue
```

Comment:
This means it can fail silently if the test framework doesn't test your case.
I would do the `for` loop param depending on the `mode` of `_assert_tests` instead